ui/components/failed.jsx

```diff
@@ -0,0 +1,173 @@
+import _ from 'lodash';
+import React from 'react';
+import ImageDiff from 'react-image-diff';
+import classNames from 'classnames';
+import {Clearfix, ButtonGroup, Button,
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+function _renderDefaultView(paths) {
+  return <div className="simple">
+    <img className="diff"
+         src={paths.diff}
+         key={paths.diff} />
+  </div>;
+}
+
+function _render2UpView(paths) {
+  return <div className="simple">
+    <Grid>
+      <Row className="show-grid">
+        <Col xs={12} md={6}>
+          <img className="baseline"
+               src={paths.baseline}
+               key={paths.baseline} />
+        </Col>
+        <Col xs={12} md={6}>
+          <img className="screenshot"
+               src={paths.screenshot}
+               key={paths.screenshot} />
+        </Col>
+      </Row>
+    </Grid>
+  </div>;
+}
+
+function _renderSwipeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="swipe"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _renderFadeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="fade"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _getSelectViewButtons(currentView, viewOptions, onViewChange) {
```

Comment:
Why isn't this function in the class' prototype? It's not used outside of it as far as I can tell.